packages/browser-wallet/src/shared/utils/token-helpers.ts

```diff
@@ -147,12 +164,23 @@ function confirmString(field?: any) {
 export const getMetadataUnique = ({ unique }: TokenMetadata) => Boolean(unique);
 export const getMetadataDecimals = ({ decimals }: TokenMetadata) => Number(decimals ?? 0);
 
+/**
+ * Given a fetch response, return the hex-encoded sha256 hash of the body.
+ */
+async function getHashOfResponse(resp: Response): Promise<string> {
+    const rawBody = [new Uint8Array(await resp.arrayBuffer())];
+    return sha256(rawBody).toString('hex');
+}
+
 /**
  * Fetches token metadata from the given url
  */
-export async function getTokenMetadata(tokenUrl: string, network: NetworkConfiguration): Promise<TokenMetadata> {
-    if (!isMainnet(network) && tokenUrl.includes('https://some.example/token/')) {
-        const id = tokenUrl.split('https://some.example/token/')[1]?.toLowerCase() ?? 'fallback';
+export async function getTokenMetadata(
+    { url, hash: checksumHash }: MetadataUrl,
+    network: NetworkConfiguration
+): Promise<TokenMetadata> {
+    if (!isMainnet(network) && url.includes('https://some.example/token/')) {
```

Comment:
I see this was already there before, but can you explain what the purpose of this function is? It seems awfully ad-hoc behaviour.

Reply:
Yes, this is very ad-hoc, I think it was mostly for development, so that we could add example tokens without actually hosting the metadata. (And so the old wCCD actually worked, it had `https://some.example/token/wccd` as metadata url)
But maybe now is the time to remove it, as it could confuse/mislead "developers" testing their tokens on testnet (if they _are_ broken)

Reply:
Yeah. It is always better to fail I think then trying to make it work in an ad-hoc manner that works some of the time. So will you remove this?


I realize that's not the Javascript way, but it should at least be our way :)